x-pack/plugins/infra/public/hooks/use_kibana_timefilter_time.tsx

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useCallback } from 'react';
+import { useUpdateEffect, useMount } from 'react-use';
+import { useKibanaContextForPlugin } from './use_kibana';
+import { TimeRange } from '../../../../../src/plugins/data/public';
+
+export const useKibanaTimefilterTime = ({
+  from: fromDefault,
+  to: toDefault,
+}: TimeRange): [typeof getTime, typeof setTime] => {
+  const { services } = useKibanaContextForPlugin();
+
+  const getTime = useCallback(() => {
+    return services.data.query.timefilter.timefilter.getTime({ from: fromDefault, to: toDefault });
+  }, [services.data.query.timefilter.timefilter, fromDefault, toDefault]);
+
+  const setTime = useCallback(
+    (timeRange: TimeRange) => {
+      services.data.query.timefilter.timefilter.setTime(timeRange);
+    },
+    [services.data.query.timefilter.timefilter]
+  );
+
+  return [getTime, setTime];
+};
+
+export const useSyncKibanaTimeFilterTime = (defaults: TimeRange, currentTimeRange: TimeRange) => {
+  const [_, setTime] = useKibanaTimefilterTime(defaults);
```

Comment:
Maybe this would satisfy the linter?

```suggestion
  const [, setTime] = useKibanaTimefilterTime(defaults);
```

Reply:
Yeah, I'll appease TS now we're happy with the implementation.